engine/runtime/src/main/scala/org/enso/interpreter/runtime/IrToTruffle.scala

```diff
@@ -2618,4 +2588,43 @@ class IrToTruffle(
 
   private def scopeAssociatedType =
     scopeBuilder.asModuleScope().getAssociatedType
+
+  private def rootScopeInfo(
+    where: => String,
+    ir: IR
+  ): () => AliasMetadata.RootScope = {
+    def readScopeInfo() = {
+      val raw =
+        ir.unsafeGetMetadata(AliasAnalysis, s"No root scope for ${where}.")
+      val scope = raw.unsafeAs[AliasMetadata.RootScope]
+
+      val log = context.getLogger()
+      if (log.isLoggable(Level.FINEST)) {
+        val allDefs = scope.graph.rootScope.allDefinitions
+        log.log(Level.FINEST, s"Scope for ${where} loaded with {0}", allDefs)
+      }
+      scope
+    }
+    readScopeInfo
+  }
+
+  private def childScopeInfo(
+    where: => String,
+    ir: IR
+  ): () => AliasMetadata.ChildScope = {
+    def readScopeInfo() = {
+      val raw =
+        ir.unsafeGetMetadata(AliasAnalysis, s"No root scope for ${where}.")
+      val scope = raw.unsafeAs[AliasMetadata.ChildScope]
+
+      val log = context.getLogger()
+      if (log.isLoggable(Level.FINEST)) {
```

Comment:
To track _materializaton_ of `LocalScope` - e.g. reading of `AliasAnalysis` metadata one can:
```
$ enso --run test/Benchmarks/src/Startup/Hello_World.enso --log-level trace 2>&1 | grep "] Scope"
```
e.g. enable most verbose logging `--log-level trace` and then search for messages that start with _"Scope"_.

Reply:
`LocalScope` is in `runtime-compiler` and that doesn't have access to `TruffleLogger`. The logger has to be provided by those who use `LocalScope`.

Reply:
`runtime-compiler` already transitively depends on `org.slf4j:slf4j-api`. There is nothing stopping you to create SLF4J logger directly in `LocalScope`.